nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,631 @@
+                cv_pred.setdefault(other_class, []).append(best_y['inverse'])
+                coefs.setdefault(other_class, []).append(-best_coef)
+                intercepts.setdefault(other_class, []).append(-best_intercept)
+
+        # TODO separe each fold, just to calculate the mean and std of the cv
+        if is_classification_:
+            classes = self.classes_
+            y_prob = np.vstack([np.hstack(cv_pred[c]) for c in classes]).T
+            self.cv_y_pred_ = self.classes_[np.argmax(y_prob, axis=1)]
+            self.cv_y_true_ = np.hstack(cv_true[cv_true.keys()[0]])
+        else:
+            classes = classes_to_predict
+            self.cv_y_pred_ = np.vstack([
+                np.hstack(cv_pred[c]) for c in classes]).T.ravel()
+            self.cv_y_true_ = np.vstack([
+                np.hstack(cv_true[c]) for c in classes]).T.ravel()
+            self.cv_params_['beta'] = self.cv_params_.pop(None)
+
+        self.coef_ = np.vstack([np.mean(coefs[c], axis=0) for c in classes])
+        self.std_coef_ = np.vstack([np.std(coefs[c], axis=0) for c in classes])
+        self.intercept_ = np.hstack([np.mean(intercepts[c], axis=0)
+                                     for c in classes])
+        coef_img = {}
+        std_coef_img = {}
+        for c, coef, std in zip(classes, self.coef_, self.std_coef_):
+            c = 'beta' if c is None else c
+            coef_img[c] = self.masker_.inverse_transform(coef)
+            std_coef_img[c] = self.masker_.inverse_transform(std)
+
+        self.coef_img_ = coef_img
+        self.std_coef_img_ = std_coef_img
+        # TODO add cv_score, like in cross_val_score, using cv_y_pred and
+        # cv_y_true (we have the results, scorer, etc)
+        # self.cv_score
+
+    def decision_function(self, niimgs, index=None):
+        """Provide prediction values for new X which can be turned into
+        a label by thresholding
+        """
+        if index is not None:
+            if isinstance(index, collections.Iterable):
+                niimgs = index_img(niimgs, index)
+        X = self.masker_.transform(niimgs)
+        if isinstance(X, tuple):
+            X = np.vstack(X)
+        X_view = X.view()
+        X_view.shape = (X.shape[0], -1)
+        decision_values = np.dot(X_view, self.coef_.T) + self.intercept_
+        return decision_values
+
+    def predict(self, niimgs, index=None):
+        """Predict a label for all X vectors indexed by the first axis"""
+        decision_values = self.decision_function(niimgs, index)
+        if self.is_classification_:
+            decisions = decision_values.argmax(axis=1)
+            decision_labels = self.classes_[decisions]
+            return decision_labels
+        return decision_values
+
+    def score(self, niimgs, y):
+        scorer, _ = _check_scorer(self, self.scoring, self.pos_label, y)
+        return scorer(self, niimgs, y)
+
+
+def _parallel_estimate(estimator, X, y, train, test, param_grid,
```

Comment:
perhaps consider putting it in front of the class, given that is a helper function used in that class